add `align` option

nexdrew opened the pull request:
Add an `align` option that supports string values `'center'` and `'right'`, allowing consumers to center- or right-align their text within the box without any pre-padding logic.

My hope is to add center-alignment to `update-notifier` "out-of-the-box" (pun intended) just by passing `align: 'center'` to `boxen`.

Here are examples (with `padding: 1`):

<img width="202" alt="screen shot 2016-06-06 at 12 54 42 pm" src="https://cloud.githubusercontent.com/assets/1929625/15830378/e73d82aa-2be5-11e6-917b-489382a057f1.png">

sindresorhus commented on index.js:
```diff
@@ -99,6 +100,15 @@ module.exports = function (text, opts) {
+		text = ansiAlign(text, {
+			align: opts.align,
+			split: NL,
+			pad: PAD
```
No need to specify the `split` and `pad` as they're the default.

nexdrew replied:
@sindresorhus True, I was just being paranoid about depending on that assumption, in case the defaults for `ansi-align` ever change.

Since that seems unlikely, I'll go ahead and change this.

sindresorhus replied:
Changing defaults would be a breaking change and require a major release anyways though.

sindresorhus commented on readme.md:
```diff
@@ -147,6 +147,13 @@ Values: `black` `red` `green` `yellow` `blue` `magenta` `cyan` `white`
+##### align
+
+Type: `string`<br>
+Values: `center` `right`
```
For completeness, also add `left` and document it as the default value.

nexdrew replied:
Will do.

nexdrew replied:
@sindresorhus Any concern that users would expect `align: 'left'` to "undo" left-padding? Because currently `ansi-align` doesn't do that. 😅

SamVerschueren replied:
I don't see why anyone would expect that behaviour?

sindresorhus replied:
Nah, it's fine.

marcjansen commented:
Nice upcoming option 👍

sindresorhus commented on index.js:
```diff
@@ -99,6 +100,11 @@ module.exports = function (text, opts) {
+
+	if (opts.align && opts.align !== 'left') {
+		text = ansiAlign(text, {align: opts.align});
+	}
```
Would be nice if `ansiAlign` could just handle the `left` option too. That way we could just do a straight passthrough:

```js
text = ansiAlign(text, {align: opts.align});
```

No `if` statement.

nexdrew replied:
Ok - fortunately I know the author. 😄 My original rationale was "if you don't need center- or right- alignment, then don't call `ansi-align`", but I see now that was probably a bit short-sighted. Let me see what I can do.

nexdrew replied:
@sindresorhus Addressed with commit 72bf8b1.

nexdrew commented:
@sindresorhus @SamVerschueren @marcjansen Thanks for the review! I believe I have addressed all comments thus far. Let me know if you see anything else.

P. S. I'm sure you're quite familiar with Angular commit conventions - the reason I use them is for auto version bumping and auto changelog generation via [`standard-version`](https://github.com/conventional-changelog/standard-version). I don't want to disrupt your current release process, but if you have time, take a look and let the [conventional-changelog team](https://github.com/orgs/conventional-changelog/people) know what you think. Thanks! ❤️

SamVerschueren commented:
@nexdrew Thanks for the PR!

We use [np](https://github.com/sindresorhus/np) for publishing our modules.

nexdrew commented:
@SamVerschueren Cool. Although I love the benefit of auto version bumping and auto changelog generation, I appreciate other tools and respect what works well for you. Thanks for sharing!

SamVerschueren commented:
@nexdrew We had a discussion regarding [generating a changelog](https://github.com/sindresorhus/np/issues/18) but it didn't look that high of a priority for anyone (yet).

sindresorhus commented:
@nexdrew I like the idea in theory, but in practice it creates more overhead than it saves. At least for me. Most contributors never read contributing.md, so I'm forced having to manually educate them about commit guidelines every single time. I've also yet to see an auto-generated changelog that is better than a clean commit history.

sindresorhus commented:
Thank you! Excellent addition :)

marcjansen commented:
👍👏👏

nexdrew commented:
@sindresorhus Thanks for merging!

Concerning this:

> Most contributors never read contributing.md, so I'm forced having to manually educate them about commit guidelines every single time.

This is a good point, but you really only need to follow a commit convention _when merging_, assuming use of "squash and merge", which is under the merger's control and can now be done via GitHub UI. And generally it's as easy as prepending `fix:` or `feat: ` to the squashed commit message (unless it's a breaking change).

> I've also yet to see an auto-generated changelog that is better than a clean commit history.

This is also a good point, but the premise of `standard-version` is that you get both - a clean commit history due to "squash and merge" and an auto-generated changelog that only includes fixes, features, and breaking changes. And the icing on the cake is auto semver bumps!

Sorry, I don't mean to drag this out. If it's ok with you, I will create an issue against `np` that documents a proposal to combine the super powers of `standard-version` and `np` so we can continue the discussion there. Thanks again!

sindresorhus commented:
I get it, but I'm not really interested in taking `np` in that direction.
